### ISSUE TITLE
git push hangs when pushing from remote

### ISSUE THREAD

**Issue by nickblock** [microsoft/vscode#219570]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 
- Version: 1.91.0-insider
Commit: aea213b7fcc7de5c24ad797ac1af209b159d451f
Date: 2024-06-28T05:55:45.367Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Linux x64 6.5.6-300.fc39.x86_64

(has had same behaviour for a number of versions now)

- OS Version: Linux -> remote into windows

Steps to Reproduce:

1. Open vscode on linux
2. Attach to remote windows machine via ssh (windows running openssh)
3. open powershell terminal in vscode
4. run git push to remote ssh linux machine using keys

command hangs

see ssh log:
[gitlog.txt](https://github.com/user-attachments/files/16057017/gitlog.txt)

git version 2.44.0.windows.1

Same command 'git push' works fine when sshing into windows machine and running from powershell, the only difference seems to be that powershell is started from within vscode.

It is also possible to log into the machine that has the git repo on it from windows powesehell in vscode without password using key.

**Comment by lszomoru** [comment]
@nickblock, sorry for not getting back to you on this until now. 
Are you able to reproduce the issue with the latest VS Code Insiders release?

**Comment by nickblock** [comment]
I printed out the env from the 2 different powershell prompts (dir env:). In vscode terminal only I see these:

VSCODE_GIT_ASKPASS_MAIN        c:\Users\nick\.vscode-server-insiders\cli\servers\Insiders-22e3447b4b410dcf8426bf70743c5dd4030d6122\server\extens...
VSCODE_GIT_ASKPASS_NODE        C:\Users\nick\.vscode-server-insiders\cli\servers\Insiders-22e3447b4b410dcf8426bf70743c5dd4030d6122\server\node.exe 
VSCODE_GIT_IPC_HANDLE          \\.\pipe\vscode-git-041154424d-sock                                                                                 
VSCODE_INJECTION               1                                                                                                                   
VSCODE_IPC_HOOK_CLI            \\.\pipe\vscode-ipc-093266a0-0205-4da4-b655-951f3d84a62e-sock     

I unset them all but it didnt help

**Comment by lszomoru** [comment]
@nickblock, do you have the possibility to remote desktop onto the Windows machine and launch VS Code locally? I just want to rule out the "Remote SSH" extension from the equation in an effort to try to narrow down the issue. Thanks!

**Comment by nickblock** [comment]
Yes, its fine just using git push from terminal within vscode just using it on windows directly.
